src/tls.c

```diff
@@ -2305,6 +2305,337 @@ int TLSX_SNI_GetFromBuffer(const byte* clientHello, word32 helloSz,
 
 #endif /* HAVE_SNI */
 
+/******************************************************************************/
+/* Trusted CA Key Indication                                                  */
+/******************************************************************************/
+
+#ifdef HAVE_TRUSTED_CA
+
+/** Creates a new TCA object. */
+static TCA* TLSX_TCA_New(byte type, const byte* id, word16 idSz, void* heap)
+{
+    TCA* tca = (TCA*)XMALLOC(sizeof(TCA), heap, DYNAMIC_TYPE_TLSX);
```

Comment:
Is a NULL id valid?

Reply:
If NULL makes it to this point it should be valid. In the case of preshared CA certs, there isn't an ID to check.